internal/restic/snapshot_policy_test.go

```diff
@@ -251,9 +255,11 @@ func TestApplyPolicy(t *testing.T) {
 					len(keep)+len(remove), len(testExpireSnapshots))
 			}
 
-			if p.Sum() > 0 && len(keep) > p.Sum() {
-				t.Errorf("not enough snapshots removed: policy allows %v snapshots to remain, but ended up with %v",
-					p.Sum(), len(keep))
+			if p.Last >= 0 && p.Hourly >= 0 && p.Daily >= 0 && p.Weekly >= 0 && p.Monthly >= 0 && p.Yearly >= 0 {
+				if p.Sum() > 0 && len(keep) > p.Sum() {
```

Comment:
`ExpirePolicy.Sum()` is only used by the tests. Thus, please convert that method into a function and move it into the tests. Afterwards, change `Sum()` to return 0 if any of the counts is `-1`.